sys/xtimer/xtimer_core.c

```diff
@@ -190,11 +194,16 @@ int _xtimer_set_absolute(xtimer_t *timer, uint32_t target)
 
     timer->target = target;
     timer->long_target = _long_cnt;
+
+    /* Ensure timer is fired in right hardware timer period. */
+    target = target - XTIMER_OVERHEAD;
```

Comment:
What happens when target is small (`target < XTIMER_OVERHEAD`)?

Reply:
Then the target underflows, it will be bigger than now and no timer overflow will happen, no long target increase needed.
Unless now+overhead > target, which should be seldome. But, this leads me to a condition which schould have been adressed above when checking for backoff.
something like this:
```
    /* first comparison prevents overflowed target to trigger backoff, target in next period.
     * second comparison backoff condition.
     * Third condition check for overflowed target
     * Fourth condition backoff condition. */
    if ( (target >= now) && ((target - XTIMER_BACKOFF) < now) ||
         (target < now) && (target <= (now + XTIMER_BACKOFF))) {

...
    /* Ensure timer is fired in right hardware timer period.
     * Backoff condition check above ensures that taget - XTIMER_OVERHEAD > now
     * also for values when now will overflow soon and target is overflowed */
    target = target - XTIMER_OVERHEAD;
```

I will have a deeper look into this tomorrow and also make some tests.

Reply:
Testing for `taget - XTIMER_OVERHEAD > now` is allready made in `_xtimer_set()` or `_xtimer_periodic_wakeup` by testing that `offset < XTIMER_BACKOFF`.

If the target for `_xtimer_set_absolute(...)` is set to close to `now` and `now` overleaps target this leads to a whole 32bit cicle delay. But this has to be ensure by the calling functions, which it is.

I added some coments to the code.